lib/reply.js

```diff
@@ -138,9 +138,30 @@ Reply.prototype.send = function (payload) {
 
   if (this[kReplySerializer] !== null) {
     payload = this[kReplySerializer](payload)
-  } else if (hasContentType === false || contentType.indexOf('application/json') > -1) {
-    if (hasContentType === false || contentType.indexOf('charset') === -1) {
+
+  // The indexOf below also matches custom json mimetypes such as 'application/hal+json' or 'application/ld+json'
+  } else if (hasContentType === false || contentType.indexOf('json') > -1) {
+    if (hasContentType === false) {
       this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+    } else {
+      // If hasContentType === true, we have a JSON mimetype
+      if (contentType.indexOf('charset') === -1) {
+        // If we have simply application/json instead of a custom json mimetype
+        if (contentType.indexOf('/json') > -1) {
+          this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
+        } else {
+          const currContentType = this[kReplyHeaders]['content-type']
+          // We extract the custom mimetype part (e.g. 'hal+' from 'application/hal+json')
+          const customJsonType = currContentType.substring(
+            currContentType.indexOf('/'),
+            currContentType.indexOf('json') + 4
```

Comment:
This should be the same

```suggestion
            currContentType.length
```

Reply:
Hi @Eomm. I added the `+ 4` to ensure we have a valid mimetype, e.g., if the user types a `;` in the end of the mimetype but do not set the charset: `application/hal+json;`. I thought it was not so costly to do that here. What do you think about it? I could also update the test cases to make this assumption clear.

Reply:
You are right, nevermind my suggestion 👍🏼